File: Front-end/src/app/models/Trip.ts
import { User } from '@auth0/auth0-angular';
import { Activity } from './Activity';
import { TripCategory } from './TripCategory';

export interface Trip {
  tripId: number;
  name: string;
  startDate: Date;
  endDate: Date;
  picture: string;
  description: string;
  city: string;
  country: string;
  isShared: boolean;
  activities: Activity[];
  categories: TripCategory[];
}

File: Back-end/TripPlanner.DAL/Migrations/Models/Trip.cs
namespace TripPlanner.DAL.Models
{
    public class Trip
    {
        public int TripId { get; set; }
        public String Name { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string? Picture { get; set; }
        public string? Description { get; set; }
        public string? Country { get; set; }
        public string? City { get; set; }
        public Boolean IsShared { get; set; }
        public ICollection<UserTrip>? Users { get; }
        public ICollection<TripActivity>? TripActivities { get; }
        public ICollection<TripKeyword>? TripKeywords { get; }
        public ICollection<TripCategory>? TripCategories { get; }
    }
}
